src/http/mime_types.cc

```diff
@@ -40,7 +46,7 @@ struct mapping {
         { "proto", "application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily; encoding=delimited"},
 };
 
-const char* extension_to_type(const sstring& extension)
+const char* extension_to_type(std::string_view extension)
 {
     for (mapping m : mappings) {
         if (extension == m.extension) {
```

Comment:
If we are already changing the function signature (BTW, it is not under `experimental` namespace) maybe then we will change also the return type and move the 
```
if (content_type.find('/') == std::string_view::npos) {
            content_type = http::mime_types::extension_to_type(content_type);
        }
        _headers["Content-Type"] = sstring(content_type);
```
part here? something like
```
sstring extension_to_type(std::string_view extension)
{
    if (extension.find('/') == std::string_view::npos) {
        for (mapping m : mappings) {
            if (extension == m.extension) {
                return m.mime_type;
            }
        }
        return "text/plain";
    }
    return sstring(extension);
}
```

Reply:
But what's the benefit? I didn't really want to change extension_to_type, I wanted to change its caller - to get std::string_view instead of sstring&.
In any case, the caller converts the returned char* to sstring, so what's the benefit of doing this inside extension_to_type()?

Reply:
> But what's the benefit? I didn't really want to change extension_to_type, I wanted to change its caller - to get std::string_view instead of sstring&. In any case, the caller converts the returned char* to sstring, so what's the benefit of doing this inside extension_to_type()?

just to save on code/comment duplicate in xxx::set_content_type. but I dont insist

Reply:
I didn't understand what you mean... We need the long comment in front of set_content_type() because it's public API and I want to explain to users what it does. I can't move that comment to some inner function that nobody uses directly.

The 4 lines of code common to both set_content_type9() I agree is duplicated, but given that the entire request and reply classes are near duplicates (but not exact...), so I don't think this is a problem.